griptape/drivers/image_generation/openai_image_generation_driver.py

```diff
@@ -86,7 +86,30 @@ def try_text_to_image(self, prompts: list[str], negative_prompts: Optional[list[
     def try_image_variation(
         self, prompts: list[str], image: ImageArtifact, negative_prompts: Optional[list[str]] = None
     ) -> ImageArtifact:
-        raise NotImplementedError(f"{self.__class__.__name__} does not support variation")
+        if self.model != "dall-e-2":
```

Comment:
Nit: Should we consolidate the model/resolution checks into a common fn? Do we see image variation & inpainting having identical supported implementation long-term or are they just identical for now?

Reply:
Consolidated. The two implementations are pretty similar, but not quite identical.